### PR TITLE
344 сreate integration tests for airdropper

### PR DESCRIPTION
**Pull request by ivandzen**

### CODE REVIEW

**Review by otselnik — DISMISSED** [review]

File: proxy/testing/test_airdropper_integration.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        print("NEON balance is: ", eth_balance)
+        self.assertTrue(eth_balance > 0 and eth_balance < 10 * pow(10, 18))  # 10 NEON is a max airdrop amount
+
+    def test_success_airdrop_complex_case(self):
+        from_owner = self.create_sol_account()
+        mint_amount = 1000_000_000_000
+        from_spl_token_acc = self.create_token_account(from_owner.public_key(), mint_amount)
+        to_neon_acc1 = self.create_eth_account().address
+        to_neon_acc2 = self.create_eth_account().address
+        sleep(15)
```
Comment (s-medvedev) [inline proxy/testing/test_airdropper_integration.py]: It needs to investigate while we need to specify this timeout.

File: proxy/testing/test_airdropper_integration.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                                                                123456))
+        trx.add(self.wrapper.create_input_liquidity_instruction(from_owner.public_key(),
+                                                                from_spl_token_acc,
+                                                                to_neon_acc2,
+                                                                654321))
+
+        opts = TxOpts(skip_preflight=True, skip_confirmation=False)
+        print(self.solana_client.send_transaction(trx, from_owner, opts=opts))
+
+        sleep(45)
```
Comment (s-medvedev) [inline proxy/testing/test_airdropper_integration.py]: Let change this large timeout with waiting in cycle with small timeouts.

File: proxy/docker-compose-test.yml [hunk 3]
```diff
@@ -82,6 +82,77 @@ services:
...
+      interval: 5s
+      timeout: 10s
+      retries: 10
+      start_period: 5s
+
+  airdropper:
+    container_name: airdropper
+    image: neonlabsorg/proxy:${REVISION}
+    environment:
+      AIRDROPPER_MODE: 'true'
```
Comment (s-medvedev) [inline proxy/docker-compose-test.yml]: I think `AIRDROPPER_MODE` should be specified in `run-airdropper.sh` file. If we need some specific tasks for airdropper, which should be done to prepare the environment (like get SOL balance, or minting some tokens), it should be specified in `run-test-airdropper.sh` script.

File: proxy/testing/test_airdropper_integration.py (hunk 1 above)
Comment (ivandzen) [reply proxy/testing/test_airdropper_integration.py]: Fixed. Delays removed

File: proxy/testing/test_airdropper_integration.py (hunk 2 above)
Comment (ivandzen) [reply proxy/testing/test_airdropper_integration.py]: Fixed

File: proxy/docker-compose-test.yml (hunk 3 above)
Comment (ivandzen) [reply proxy/docker-compose-test.yml]: Fixed

File: run-test-faucet.sh [hunk 4]
```diff
@@ -1,7 +1,44 @@
...
+    echo "airdropping..."
+    solana airdrop 5000 "$ACCOUNT"
+    # check that balance >= 10 otherwise airdroping by 1 SOL up to 10
+    BALANCE=$(solana balance "$ACCOUNT" | tr '.' '\t'| tr '[:space:]' '\t' | cut -f1)
+    while [ "$BALANCE" -lt 10 ]; do
+      solana airdrop 1 "$ACCOUNT"
+      sleep 1
+      BALANCE=$(solana balance "$ACCOUNT" | tr '.' '\t'| tr '[:space:]' '\t' | cut -f1)
+    done
+fi
```
Comment (s-medvedev) [inline run-test-faucet.sh]: Maybe better to use: `create-test-accounts.sh 1`?

**Review by s-medvedev — DISMISSED** [review]

**Review by s-medvedev — APPROVED** [review]

File: run-test-faucet.sh (hunk 4 above)
Comment (ivandzen) [reply run-test-faucet.sh]: Fixed

**Pull request merged**